src/Views/FirmwareView.vala

```diff
@@ -148,4 +172,116 @@ public class About.FirmwareView : Gtk.Stack {
 
         update_list.show_all ();
     }
+
+    private void show_release (Gtk.ListBoxRow widget) {
+        if (widget is Widgets.FirmwareUpdateRow) {
+            var row = (Widgets.FirmwareUpdateRow) widget;
+            firmware_release_view.update_view (row.device, row.device.latest_release);
+            deck.visible_child = firmware_release_view;
+
+            firmware_release_view.back.connect (() => {
+                deck.navigate (Hdy.NavigationDirection.BACK);
+            });
```

Comment:
This can probably be connected in the construct block. It doesn't really make sense to connect this every time the view contents change since the back button will never change